lib/cisco_node_utils/bridge_domain_vni.rb

```diff
@@ -0,0 +1,206 @@
+# NXAPI implementation of BridgeDomainVNI class
+#
+# February 2016, Rohan Gandhi Korlepara
+#
+# Copyright (c) 2015-2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+require_relative 'feature'
+
+module Cisco
+  # node_utils class for bridge_domain_vni
+  class BridgeDomainVNI < NodeUtil
+    attr_reader :bd_ids, :bd_ids_list
+
+    def initialize(bds, instantiate=true)
+      # Spaces are removed as bridge-domain cli doesn't accept value with
+      # space
+      @bd_ids = bds.to_s.gsub(/\s+/, '')
+      fail 'bridge-domain value is empty' if @bd_ids.empty? # no empty strings
+
+      @bd_ids_list = BridgeDomainVNI.string_to_array(@bd_ids)
+      create if instantiate
+    end
+
+    def to_s
+      "Bridge Domain #{bd_ids}"
+    end
```

Comment:
This is for the CliErrors raised by any NodeUtil to avoid the cli_error_check function